RankUpper-56/src/main/java/br/net/fabiozumbi12/RankUpper/RUCommands.java

```diff
@@ -438,31 +474,27 @@ private static void SendCheckMessage(CommandSource sender, User playerToCheck) {
             Optional<PlaceholderService> phapiOpt = Sponge.getServiceManager().provide(PlaceholderService.class);
             if (phapiOpt.isPresent()) {
                 PlaceholderService phapi = phapiOpt.get();
-                for (Entry<String, Long> key:RankUpper.get().getConfig().root().ranked_groups.get(pgroup).placeholder_api_requirements.entrySet()){
-                    Optional<Long> optVal = phapi.parse(key.getKey(), playerToCheck.getPlayer().isPresent() ? playerToCheck.getPlayer().get(): playerToCheck, null, Long.class);
-                    if (optVal.isPresent() && phapiOpt.get().isRegistered(key.getKey())){
-                        long needed = key.getValue();
-                        long actual = optVal.get();
-                        if (actual >= key.getValue()){
-                            sender.sendMessage(RUUtil.toText(RankUpper.get().getLang().get("config.placeholderapi").replace("{placeholder}", key.getKey().replace("%", "")) + ": &a"+actual+"/"+needed + " - " + RankUpper.get().getLang().get("config.ok")));
-                        } else {
-                            sender.sendMessage(RUUtil.toText(RankUpper.get().getLang().get("config.placeholderapi").replace("{placeholder}", key.getKey().replace("%", "")) + ": &c"+actual+"/"+needed));
-                        }
+                for (Entry<String, Long> key : RankUpper.get().getConfig().root().ranked_groups.get(pgroup).placeholder_api_requirements.entrySet()) {
+                    double optVal = RUUtil.getPlaceholderValue(phapi, key.getKey(), playerToCheck.getPlayer().isPresent() ? playerToCheck.getPlayer().get() : playerToCheck);
+                    long needed = key.getValue();
+                    if (optVal >= key.getValue()) {
+                        sender.sendMessage(RUUtil.toText(RankUpper.get().getLang().get("config.placeholderapi." + key.getKey() + ".ok").replace("{current}", optVal + "").replace("{target}", needed + "")));
```

Comment:
These were the only changes in this file, the rest if intellij auto formatting the class